src/dbnode/storage/block/block.go

```diff
@@ -223,6 +231,15 @@ func (b *dbBlock) Stream(blocker context.Context) (xio.BlockReader, error) {
 		return stream, nil
 	}
 
+	b.RUnlock()
+	lockUpgraded = true
```

Comment:
Btw, maybe we should rename the lowercase `stream(...)` to `streamWithRLock(...)`?

Reply:
sure thing